File: app/src/main/java/com/aref/jarvis/ContactHelper.java
package com.aref.jarvis;
import android.content.*;import android.database.Cursor;import android.provider.ContactsContract;
public final class ContactHelper{
 private ContactHelper(){}
 public static String name(Context c,String n){try{Cursor x=c.getContentResolver().query(ContactsContract.PhoneLookup.CONTENT_FILTER_URI,new String[]{ContactsContract.PhoneLookup.DISPLAY_NAME},ContactsContract.PhoneLookup.NUMBER+" LIKE ?",new String[]{"%"+n.replaceAll("[^0-9+]","")+"%"},null);if(x!=null)try{return x.moveToFirst()?x.getString(0):"";}finally{x.close();}}catch(Exception ignored){}return "";}
}

File: app/src/main/java/com/example/service/ContactHelper.kt
package com.example.service

import android.content.Context
import android.database.Cursor
import android.net.Uri
import android.provider.ContactsContract

object ContactHelper {

    fun getContactName(context: Context, phoneNumber: String): String? {
        if (phoneNumber.isBlank()) return null
        return try {
            val uri = Uri.withAppendedPath(
                ContactsContract.PhoneLookup.CONTENT_FILTER_URI,
                Uri.encode(phoneNumber)
            )
            val projection = arrayOf(ContactsContract.PhoneLookup.DISPLAY_NAME)
            val cursor: Cursor? = context.contentResolver.query(uri, projection, null, null, null)
            cursor?.use {
                if (it.moveToFirst()) {
                    val nameIndex = it.getColumnIndex(ContactsContract.PhoneLookup.DISPLAY_NAME)
                    if (nameIndex != -1) {
                        return it.getString(nameIndex)
                    }
                }
            }
            null
        } catch (_: Exception) {
            null
        }
    }
}
